model/minio/minio_server.rb

```diff
@@ -62,9 +62,18 @@ def init_health_monitor_session
     }
   end
 
+  def server_data(client = self.client)
+    server_data = JSON.parse(client.admin_info.body)["servers"]
+    if cluster.server_count == 1
+      server_data.first
+    else
+      server_data.find { it["endpoint"] == endpoint }
+    end
```

Comment:
Is this branch necessary? It seems the number of comparisons avoided in the fast path (count == 1) is quite tiny. Is it the case that we don't want to validate endpoint equality because of the cardinality as well?

Reply:
Yes, it is necessary. For single server Minio clusters, `it["endpoint"] != endpoint`, as `it["endpoint"]` does not use the `index` as a suffix (unlike with multiple server Minio clusters).